### PR TITLE
+doc #20192 explain need of draining entities in server/client HTTP

### PR DESCRIPTION
**Pull request by ktoso**
As highlighted by https://github.com/akka/akka/issues/20192 and numerous other questions in support and chat etc, we have not done a good job explaining that draining entities is a requirement.

People would copy & paste our examples, and be confused why it's not working – problem is, even our examples did not do the right thing (as they were old + ported from spray + and never really updated properly in recent memory).

This PR contains docs explaining the need and some ways to do the draining/discarding, and also links to it. Some of the docs examples which were doing the wrong thing are fixed too.

Pending is the Java version of it, please review as is and then I'll do the Java version accordingly

### CODE REVIEW

File: akka-docs/rst/scala/code/docs/http/scaladsl/server/directives/RouteDirectivesExamplesSpec.scala [hunk 1]
```diff
@@ -22,10 +22,10 @@ class RouteDirectivesExamplesSpec extends RoutingSpec {
           complete(StatusCodes.OK)
         } ~
         path("c") {
-          complete(StatusCodes.Created, "bar")
+          complete(StatusCodes.Created -> "bar")
         } ~
         path("d") {
-          complete(201, "bar")
+          complete(201 -> "bar")
```
Comment (ktoso) [inline akka-docs/rst/scala/code/docs/http/scaladsl/server/directives/RouteDirectivesExamplesSpec.scala]: less auto tupling

File: akka-docs/rst/scala/code/docs/http/scaladsl/server/WebSocketExampleSpec.scala [hunk 2]
```diff
@@ -84,6 +86,7 @@ class WebSocketExampleSpec extends WordSpec with Matchers with CompileOnlySpec {
         .collect {
           case tm: TextMessage => TextMessage(Source.single("Hello ") ++ tm.textStream)
           // ignore binary messages
+          // TODO in case a Streamed message comes in, we should runWith(Sink.ignore) its data
```
Comment (ktoso) [inline akka-docs/rst/scala/code/docs/http/scaladsl/server/WebSocketExampleSpec.scala]: note to self, add ref to https://github.com/akka/akka/issues/20096

**Comment by ktoso** [comment]
Pushed test name typo fix so this should pass now

**Comment by ktoso** [comment]
@akka/akka-team can one have a look here? Would like to merge :)

File: akka-docs/rst/scala/http/implications-of-streaming-http-entity.rst [hunk 3]
```diff
@@ -0,0 +1,129 @@
+.. _implications-of-streaming-http-entities:
+
+Implications of the streaming nature of Request/Response Entities
+-----------------------------------------------------------------
+
+Akka HTTP is streaming *all the way through*, which means that the back-pressure mechanisms enabled by Akka Streams
+are exposed through all layers–from the TCP layer, through the HTTP server, all the way up to the user-facing ``HttpRequest`` 
+and ``HttpResponse`` and their ``HttpEntity`` APIs.
+
+This has suprising implications if you are used to non-streaming / not-reactive HTTP clients.
```
Comment (patriknw) [inline akka-docs/rst/scala/http/implications-of-streaming-http-entity.rst]: non-reactive?

**Comment by patriknw** [comment]
LGTM

**Comment by ktoso** [comment]
Will add java version (copy paste) and merge

**Comment by ktoso** [comment]
Pushed with java version

**Pull request merged**